src/Google.Api.CommonProtos/TypeExtensions/Date.cs

```diff
@@ -0,0 +1,99 @@
+﻿/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using System;
+
+namespace Google.Type
+{
+    /// <summary>
+    /// Extensions to <see cref="Date"/>.
+    /// </summary>
+    public partial class Date
+    {
+        /// <summary>
+        /// Converts <see cref="Date"/> to <see cref="DateTime"/>.
+        /// </summary>
+        /// <returns>The converted <see cref="DateTime"/> with no time specified.</returns>
```

Comment:
I'd say "at midnight" rather than "with no time specified" given that a `DateTime` *always* has a time-of-day. I think it's worth saying that the `Kind` is `Unspecified` though.